### PR TITLE
self monitoring integration tests

### PR DESCRIPTION
**Pull request by dstathis**
## Issue
Integration tests for the recently added self monitoring.

## Solution
Added integration tests

## Testing Instructions
It tests itself...

## Release Notes
-

### CODE REVIEW

**Comment by dstathis** [comment]
The diff looks kind of wonky for `helpers.py`. It may be easier to just look at the file. The function `loki_endpoint_request` was added.

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -41,8 +42,20 @@ async def loki_rules(ops_test, app_name) -> dict:
         response = urllib.request.urlopen(f"{url}/loki/api/v1/rules", data=None, timeout=2.0)
         if response.code == 200:
             return yaml.safe_load(response.read())
-        return {}
+        return ""
```
Comment (rbarry82) [inline tests/integration/helpers.py:45]: Why the changed return type when we say it's `dict`?

**Review by rbarry82 — APPROVED** [review]

File: tests/integration/test_self_monitoring.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    assert len(metrics) > 0
+
+
+async def test_query_metrics_from_prometheus(ops_test):
+    address = await get_unit_address(ops_test, PROMETHEUS, 0)
+    url = f"http://{address}:9090/api/v1/query"
+    params = {"query": f"up{{juju_application='{LOKI}'}}"}
+    try:
+        response = requests.get(url, params=params)
+        assert response.json()["status"] == "success"
```
[the remark this answers is not in the document]
Comment (dstathis) [reply tests/integration/test_self_monitoring.py:60]: We did check for a specific metric. We checked for `up`.

File: tests/integration/helpers.py (hunk 1 above)
Comment (dstathis) [reply tests/integration/helpers.py:45]: Oops I was supposed to do that change for `loki_endpoint_request`. Fixed.

**Pull request merged**